sw/device/tests/alert_handler_lpg_reset_toggle.c

```diff
@@ -0,0 +1,548 @@
+
+/**
+ * A structure to keep the info for peripheral IPs
+ */
+typedef struct test {
+  /**
+   * Name of the peripheral.
+   */
+  const char *name;
+  /**
+   * Base address for the peripheral.
+   */
+  uintptr_t base;
+  /**
+   * Handle to the DIF object for this peripheral.
+   */
+  void *dif;
+  /**
+   * List of Alert IDs for the peripheral
+   */
+  const uint32_t *alert_ids;
+  /**
+   * number of alerts for the peripheral
+   */
+  const uint32_t num_alert_peri;
+  /**
+   * The index of this device in the reset manager.
+   */
+  uint32_t reset_index;
+} test_t;
+
+// The array of the peripherals
+static const test_t kPeripherals[] = {
+    {
+        .name = "SPI_HOST0",
+        .base = TOP_EARLGREY_SPI_HOST0_BASE_ADDR,
+        .dif = &spi_host0,
+        .alert_ids = spihost0_alerts,
+        .num_alert_peri = num_spihost0_alerts,
+        .reset_index = kTopEarlgreyResetManagerSwResetsSpiHost0,
+    },
+    {
+        .name = "SPI_HOST1",
+        .base = TOP_EARLGREY_SPI_HOST1_BASE_ADDR,
+        .dif = &spi_host1,
+        .alert_ids = spihost1_alerts,
+        .num_alert_peri = num_spihost1_alerts,
+        .reset_index = kTopEarlgreyResetManagerSwResetsSpiHost1,
+    },
+    {
+        .name = "USB",
+        .base = TOP_EARLGREY_USBDEV_BASE_ADDR,
+        .dif = &usbdev,
+        .alert_ids = usbdev_alerts,
+        .num_alert_peri = num_usbdev_alerts,
+        .reset_index = kTopEarlgreyResetManagerSwResetsUsb,
+    },
+    {
+        .name = "SPI_DEVICE",
+        .base = TOP_EARLGREY_SPI_DEVICE_BASE_ADDR,
+        .dif = &spi_dev.dev,
+        .alert_ids = spidev_alerts,
+        .num_alert_peri = num_spidev_alerts,
+        .reset_index = kTopEarlgreyResetManagerSwResetsSpiDevice,
+    },
+    {
+        .name = "I2C0",
+        .base = TOP_EARLGREY_I2C0_BASE_ADDR,
+        .dif = &i2c0,
+        .alert_ids = i2c0_alerts,
+        .num_alert_peri = num_i2c0_alerts,
+        .reset_index = kTopEarlgreyResetManagerSwResetsI2c0,
+    },
+    {
+        .name = "I2C1",
+        .base = TOP_EARLGREY_I2C1_BASE_ADDR,
+        .dif = &i2c1,
+        .alert_ids = i2c1_alerts,
+        .num_alert_peri = num_i2c1_alerts,
+        .reset_index = kTopEarlgreyResetManagerSwResetsI2c1,
+    },
+    {
+        .name = "I2C2",
+        .base = TOP_EARLGREY_I2C2_BASE_ADDR,
+        .dif = &i2c2,
+        .alert_ids = i2c2_alerts,
+        .num_alert_peri = num_i2c2_alerts,
+        .reset_index = kTopEarlgreyResetManagerSwResetsI2c2,
+    },
+};
+
+/**
+ * Configure and enable alerts of `num_peripheral` peripherals starting from the
+ * `first_peripheral`.
+ * @param num_peripherals The number of peripherals, of which alerts will be
+ * enabled
+ * @param first_peripheral The address of the first peripheral in the range
+ * @param ping_timeout The timeout value for the ping timer object.
+ */
+static void alert_handler_config_peripherals(uint32_t num_peripherals,
```

Comment:
This is just a suggestion and probably does not need to be in this PR.
I wonder if this function has been used a few times in SW test.
Maybe we can consider move it to a common area?

Reply:
> This is just a suggestion and probably does not need to be in this PR. I wonder if this function has been used a few times in SW test. Maybe we can consider move it to a common area?

You are right. In the other alert_handler tests, slightly different version of this function is used to enable and lock all of the peripheral alerts. Actually, this one is a wrapper around another utility function `alert_handler_testutils_configure_all`.

As far as I understand, `alert_handler_config_peripherals` was implemented as a test-specific wrapper around the `alert_handler_testutils_configure_all`. But all alert-handler tests so far seem to be using the same/similar config. I can check the other tests and prepare a draft PR to see if we can make this a generic wrapper. 

WDYT?

Reply:
sure, sounds good, thanks Bilgiday.